util/plugins/eu.esdihumboldt.util.test/src/eu/esdihumboldt/util/geometry/WindingOrderTest.java

```diff
@@ -0,0 +1,130 @@
+import com.vividsolutions.jts.geom.LinearRing;
+import com.vividsolutions.jts.geom.MultiPolygon;
+import com.vividsolutions.jts.geom.Polygon;
+
+/**
+ * TODO Type description
+ * 
+ * @author Arun
+ */
+public class WindingOrderTest {
+
+	private static Polygon clockWise1, clockWise2, clockWise2WOHoles;
+	private static MultiPolygon clockWise3;
+
+	private static GeometryCollection clockWise4;
+
+	@BeforeClass
+	public static void setUp() {
+		GeometryFactory factory = new GeometryFactory();
+
+		LinearRing r1 = factory.createLinearRing(new Coordinate[] { new Coordinate(10, 30),
+				new Coordinate(20, 0), new Coordinate(0, 0), new Coordinate(10, 30) });
+
+		clockWise1 = factory.createPolygon(r1);
+
+		LinearRing r2 = factory
+				.createLinearRing(new Coordinate[] { new Coordinate(49.87445, 8.64729),
+						new Coordinate(49.87582, 8.65441), new Coordinate(49.87095, 8.65694),
+						new Coordinate(49.86978, 8.65032), new Coordinate(49.87197, 8.64758),
+						new Coordinate(49.87341, 8.64688), new Coordinate(49.87445, 8.64729) });
+
+		LinearRing h1 = factory
+				.createLinearRing(new Coordinate[] { new Coordinate(49.87327, 8.64991),
+						new Coordinate(49.8735, 8.6521), new Coordinate(49.87253, 8.65239),
+						new Coordinate(49.8723, 8.65045), new Coordinate(49.87327, 8.64991) });
+
+		LinearRing h2 = factory
+				.createLinearRing(new Coordinate[] { new Coordinate(49.87203, 8.65208),
+						new Coordinate(49.87209, 8.6531), new Coordinate(49.87156, 8.65312),
+						new Coordinate(49.87145, 8.65227), new Coordinate(49.87203, 8.65208) });
+
+		clockWise2 = factory.createPolygon(r2, new LinearRing[] { h1, h2 });
+
+		clockWise2WOHoles = factory.createPolygon(r2);
+
+		clockWise3 = factory.createMultiPolygon(new Polygon[] { clockWise1, clockWise2 });
+
+		clockWise4 = factory.createGeometryCollection(
+				new Geometry[] { clockWise2, clockWise2WOHoles, clockWise3, r2 });
+
+	}
+
+	@Test
+	public void testUnifyWOHoles() {
+		Geometry result = WindingOrder.unifyWindingOrder(clockWise2WOHoles, true);
+		assertTrue(result instanceof Polygon);
+		assertTrue(clockWise2WOHoles.equalsExact(result));
+	}
+
+	@Test
+	public void testOrderClockwise() {
+		assertFalse(WindingOrder.isCounterClockwise(clockWise1.getExteriorRing()));
+
+	}
+
+	@Test
+	public void testOrderCounterClockwise() {
+		assertTrue(WindingOrder.isCounterClockwise(clockWise2.getExteriorRing()));
+	}
+
+	@Test
+	public void testUnifyCCWSimple() {
+		Geometry result = WindingOrder.unifyWindingOrder(clockWise1, true);
+		assertTrue(result instanceof Polygon);
+		assertFalse(clockWise1.equalsExact(result));
+		assertTrue(WindingOrder.isCounterClockwise(((Polygon) result).getExteriorRing()));
+	}
+
+	@Test
+	public void testUnifyCCWWithHoles() {
+		Geometry result = WindingOrder.unifyWindingOrder(clockWise2, true);
+		assertTrue(result instanceof Polygon);
+		assertFalse(clockWise2.equalsExact(result));
+		assertTrue(WindingOrder.isCounterClockwise(((Polygon) result).getExteriorRing()));
```

Comment:
You added new tests for that. What I actually meant was that in this test, in addition to testing the shell of the processed polygon you could verify the winding order of the holes of the processed polygon.